config/config/src/Config.ts

```diff
@@ -133,6 +134,23 @@ export interface Config {
   useRunningStoreServer?: boolean
   workspaceConcurrency: number
   workspaceDir?: string
+  workspaceManifest?: {
+    /**
+     * The directory the pnpm-workspace.yaml file was found in.
+     *
+     * The value of this field will be equivalent to the workspaceDir config
+     * field, but is repeated here for convenience. The workspaceDir field is
+     * nullable, while this field is not, making it easier to access the
+     * directory and contents of the workspace manifest together without null
+     * checks.
+     */
+    dir: string
```

Comment:
@zkochan Adding a `workspacePackagePatterns` field to the top-level of `Config` makes it harder to write the code above.

We have to check that both `workspacePackagePatterns` and `workspaceDir` are non-null, but that's a bit of a code smell because they're supposed to either be both defined, or both undefined.

```ts
if (opts.workspaceDir != null && opts.workspacePackagePatterns != null) {
  await findWorkspacePackages(opts.workspaceDir, {
    ...opts,
    patterns: opts.workspacePackagePatterns
  })
}
```

This is a problem for the next PR in https://github.com/pnpm/pnpm/pull/8214.

Maybe we can do something like this in the `Config` object?

```ts
interface Config {
  workspace?: {
    dir: string;
    packagePatterns: string;

    // other existing fields might also make sense here
    hoistWorkspacePackages?: boolean;
  }
}
```